GDBMS_ALGO/community/louvain.gsql

```diff
@@ -78,23 +78,24 @@ CREATE TEMPLATE QUERY GDBMS_ALGO.community.louvain(
   // -------------------- 1. First pass --------------------
   // Initialization
   All_Nodes = {v_type_set};
-  Pass_Nodes = SELECT s
-               FROM   All_Nodes:s -(e_type_set:e)- :t
-               ACCUM  @@m += e.getAttr(weight_attribute, "DOUBLE") / 2,
-                      s.@k += e.getAttr(weight_attribute, "DOUBLE"),
-                      IF s == t THEN // Self-loop link
-                        s.@k_self_loop += e.getAttr(weight_attribute, "DOUBLE")
-                      END
-               POST-ACCUM
-                      s.@community_id = s,
-                      s.@vid = getvid(s),
-                      s.@batch_id = s.@vid % total_batch_count;
+  All_Nodes = SELECT s
+              FROM   All_Nodes:s -(e_type_set:e)- :t
+              ACCUM  @@m += e.getAttr(weight_attribute, "DOUBLE") / 2,
+                     s.@k += e.getAttr(weight_attribute, "DOUBLE"),
+                     IF s == t THEN // Self-loop link
+                       s.@k_self_loop += e.getAttr(weight_attribute, "DOUBLE")
+                     END
+              POST-ACCUM
+                     s.@community_id = s,
+                     s.@vid = getvid(s),
+                     s.@batch_id = s.@vid % total_batch_count;
   IF @@m < 0.00000000001 THEN
     RETURN;
   END;
 
   // Local moving
   INT hop = 0;
+  Pass_Nodes = All_Nodes;
```

Comment:
Why not just set Pass_Nodes to the resultant set on line 81 instead of reassigning/renaming here?

Reply:
Also, a key reason for the change is that to write to the file correctly (lines 380-381), All_Nodes should be limited to nodes with an outdegree > 0. Otherwise, since the @community_id of nodes without edges will not be initialized, it will cause an error on lines 380-381.